libmemscan/src/interactive.rs

```diff
@@ -0,0 +1,534 @@
+                continue; // Address no longer readable
+            }
+            
+            let current = match Value::from_bytes(&buffer, 0, self.value_type) {
+                Some(v) => v,
+                None => continue,
+            };
+            
+            let keep = match op {
+                FilterOp::Equals => {
+                    if let Some(ref val) = compare_value {
+                        values_equal(&current, val)
+                    } else {
+                        false
+                    }
+                }
+                FilterOp::LessThan => {
+                    if let Some(ref val) = compare_value {
+                        value_less_than(&current, val)
+                    } else {
+                        false
+                    }
+                }
+                FilterOp::GreaterThan => {
+                    if let Some(ref val) = compare_value {
+                        value_greater_than(&current, val)
+                    } else {
+                        false
+                    }
+                }
+                FilterOp::Increased => value_greater_than(&current, &match_entry.current_value),
+                FilterOp::Decreased => value_less_than(&current, &match_entry.current_value),
+                FilterOp::Changed => !values_equal(&current, &match_entry.current_value),
+                FilterOp::Unchanged => values_equal(&current, &match_entry.current_value),
+            };
+            
+            if keep {
+                new_matches.push(MatchedAddress {
+                    address: match_entry.address,
+                    current_value: current,
+                    previous_value: Some(match_entry.current_value.clone()),
+                });
+            }
+        }
+        
+        self.matches = new_matches;
+        
+        // Clean up regions with no matches
+        self.cleanup_empty_regions();
+        
+        Ok(self.matches.len())
+    }
+    
+    /// Remove regions that have no matching addresses
+    fn cleanup_empty_regions(&mut self) {
+        if self.matches.is_empty() {
+            self.regions.clear();
+            return;
+        }
+        
+        // Determine which regions still have matches
+        let mut active_regions = std::collections::HashSet::new();
+        for match_entry in &self.matches {
+            for (idx, region) in self.regions.iter().enumerate() {
+                if match_entry.address >= region.base_address
+                    && match_entry.address < region.base_address + region.size
+                {
+                    active_regions.insert(idx);
+                    break;
+                }
+            }
+        }
+        
+        // Keep only active regions
+        let mut new_regions = Vec::new();
+        for (idx, region) in self.regions.iter().enumerate() {
+            if active_regions.contains(&idx) {
+                new_regions.push(region.clone());
```

Comment:
Maybe do a more efficient **safe** memory operation to move it instead of cloning it.